### PR TITLE
Finance data processing

### PR DESCRIPTION
**Pull request by cvriel**
Data processing DAG for finance data for Tableau BI users.

### CODE REVIEW

**Review by Bash- — APPROVED** [review]

File: src/dags/financien_grootboek.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+from common import (
+    default_args,
+    DATAPUNT_ENVIRONMENT,
+    slack_webhook_token,
+    MessageOperator,
+)
+from common.sql import SQL_CHECK_COUNT
+
+DATASTORE_TYPE: str = (
```
Comment (gkoller) [inline src/dags/financien_grootboek.py:20]: This exact definition is already present is 7 other DAGs:

- src/dags/anpr.py
- src/dags/beschermde_stadsdorpsgezichten.py
- src/dags/blackspots.py
- src/dags/beheerkaart.py
- src/dags/rioolnetwerk.py
- src/dags/varen.py
- src/dags/woningbouwplannen.py

I like to see this definition moved to the `common` module and removed from those 7 DAG and this new DAG.

File: src/dags/financien_grootboek.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    "acceptance" if DATAPUNT_ENVIRONMENT == "development" else DATAPUNT_ENVIRONMENT
+)
+
+data_schema_id: str = "financien"
+dag_id: str = "financien_grootboek"
+
+
+with DAG(
+    dag_id,
+    default_args={**default_args},
```
Comment (gkoller) [inline src/dags/financien_grootboek.py:30]: You seem to be destructing the dictionary only to create another one. You would do that if you expect the DAG to modify the `default_args`. However it does not. On the contrary, it does a `deepcopy` which is safer than the destructuring using `**`. Hence you should change this to: `default_args=default_args`

Note: see `DAG.__init__(...)`, it does `self.default_args = copy.deepcopy(default_args or {})`

File: src/dags/financien_grootboek.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        sql=SQL_DROP_TMP_TABLE,
+        params=dict(tablename=f"{dag_id}_new"),
+    )
+
+    # 8. Grant database permissions
+    grant_db_permissions = PostgresPermissionsOperator(task_id="grants", dag_name=data_schema_id)
+
+
+# FLOW
+[
```
Comment (gkoller) [inline src/dags/financien_grootboek.py:104]: No need to create a one element list here. The parenthesis used in other DAG do not create a one element iterable either (a tuple). They are just there for structuring/formatting purposes)

File: src/dags/importscripts/import_financien_grootboek.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            ,org.RVE_NAAM as "rve_naam"
+            ,org.CLUSTER_CODE as "cluster_code"
+            ,org.CLUSTER_NAAM as "cluster_naam"
+            FROM DATAHUB.FI2_DHB_FCT_GROOTBOEK GB
+            INNER JOIN DATAHUB.FI2_DHB_DIM_RELATIE REL ON REL.ID = GB.REL_ID
+            INNER JOIN DATAHUB.FI2_DHB_DIM_ORGANISATIE org ON ORG.ID = GB.ORG_ID
+            INNER JOIN DATAHUB.FI2_DHB_DIM_GROOTBOEKREKENING gbr ON gbr.ID = GB.gbr_ID
+            INNER JOIN DATAHUB.FI2_DHB_DIM_VERPLICHTINGREGEL vpr ON vpr.ID = GB.VPR_ID
+            INNER JOIN DATAHUB.FI2_DHB_DIM_WERKORDER wor ON wor.ID = GB.WOR_ID
+            WHERE 1=1
```
Comment (gkoller) [inline src/dags/importscripts/import_financien_grootboek.py:75]: I see this often in your SQL statements. It does nothing. Might I suggest you remove it?

File: src/dags/importscripts/import_financien_grootboek.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            "rve_code": Text(),
+            "rve_naam": Text(),
+            "cluster_code": Text(),
+            "cluster_naam": Text(),
+        }
+        df.to_sql(
+            table_name, db_engine, if_exists="replace", index_label="id", index=True, dtype=dtype
+        )
+        with db_engine.connect() as connection:
+            connection.execute(f"ALTER TABLE {table_name} ADD PRIMARY KEY (ID)")
```
Comment (gkoller) [inline src/dags/importscripts/import_financien_grootboek.py:138]: We should not be using regular string formatting for constructing SQL queries. It's probably safe here, but the practice gets copied over and over and as such persists. Better to use the features of `psycopg2`: https://www.psycopg.org/docs/sql.html

Using that the statement would become (untested):
```python
from psycopg2 import sql
...
connection.execute(
    sql.SQL("ALTER TABLE {table_name} ADD PRIMARY KEY (ID)")
        .format(table_name=sql.Identifier(table_name)
)
```

**Pull request merged**